File: packages/kotlin/src/main/kotlin/com/natsuneko/catalyst/models/CatalystTrend.kt
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

package com.natsuneko.catalyst.models

import kotlinx.serialization.Serializable

/**
 * A single entry in the "rich" trending list (`GET /catalyst/v1/trend?format=rich`)
 *
 * [movement] is one of "up" | "down" | "same" | "new"
 */
@Serializable
data class CatalystRichTrendingItem(
    val tag: String,
    val rank: Int,
    val previousRank: Int? = null,
    val movement: String
)

File: packages/swift/Sources/CatalystSwift/Models/CatalystTrend.swift
// Licensed under the MIT License
//
// Copyright (c) 2025 Natsune Mochizuki

/// A single entry in the "rich" trending list (`GET /catalyst/v1/trend?format=rich`).
///
/// `movement` is one of "up" | "down" | "same" | "new".
public struct CatalystRichTrendingItem: Decodable, Sendable, Equatable, Hashable {
  public let tag: String
  public let rank: Int
  public let previousRank: Int?
  public let movement: String
}
